pal/os/inc/linux/az_pal_os.h

```diff
@@ -1,15 +1,22 @@
 // Copyright (c) Microsoft. All rights reserved.
-// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
 
 #ifndef LINUX_AZ_PAL_OS_H
 #define LINUX_AZ_PAL_OS_H
 
 #include <pthread.h>
+#include <semaphore.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ *  @struct AZ_PAL_OS_LOCK
+ *
+ *  @brief  pointer to a platform specific struct for a lock implementation
+ */
```

Comment:
this should debatably be moved back to the higher level pal. this could cause multiple document generations for doxygen for the same struct

Reply:
We cannot move it up.

---
In reply to: [342855539](https://github.com/Azure/azure-ulib-c/pull/47#discussion_r342855539) [](ancestors = 342855539)